### PR TITLE
KFS Release 0.5

### PR DESCRIPTION
**Pull request by yuzisun**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/samples/v1beta1/triton/torchscript/README.md [hunk 1]
```diff
@@ -192,16 +224,14 @@ Status Codes  [code:count]                      200:6000
 Error Set:
 ```
 
-# Add transformer step before the triton inference server
+## Add Transformer on the InferenceService
```
Comment (animeshsingh) [inline docs/samples/v1beta1/triton/torchscript/README.md:227]:
```suggestion
## Add Transformer to the InferenceService
```

File: docs/samples/v1beta1/triton/torchscript/README.md [hunk 2]
```diff
@@ -192,16 +224,14 @@ Status Codes  [code:count]                      200:6000
...
 
 `Triton Inference Server` expects tensors as input data, often the time a pre-processing step is required before making the prediction call
```
Comment (animeshsingh) [inline docs/samples/v1beta1/triton/torchscript/README.md:229]:
```suggestion
`Triton Inference Server` expects tensors as input data, often times a pre-processing step is required before making the prediction call
```

File: python/kfserving/setup.py [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 setuptools.setup(
     name='kfserving',
-    version='0.5.0',
+    version='0.5.0.1',
```
Comment (animeshsingh) [inline python/kfserving/setup.py:28]: Have we done 4 digit version numbers?

**Comment by animeshsingh** [comment]
@pvaneck can you review as well

File: python/kfserving/setup.py (hunk 3 above)
Comment (yuzisun) [reply python/kfserving/setup.py:28]: yes, we have done before.

**Comment by PatrickXYS** [comment]
@yuzisun I might know the reason why 401 Unauthorized happened, though not dig into it very deep.

As what I saw, when make API call such as 

```
  client = k8s_client.ApiClient()
  crd_api = k8s_client.CustomObjectsApi(client)
  return crd_api.get_namespaced_custom_object(
    GROUP, VERSION, namespace, PLURAL, name)
```

Every 15 mins, the attached token has to be refreshed otherwise 401 Unauthorized. 

E.g, check prow logs and you can see every 15mins, 401 Unauthorized error surfaces, http://prow.kubeflow-testing.com/view/s3/aws-kubeflow-jenkins/pr-logs/pull/kubeflow_kfserving/1315/kubeflow-kfserving-presubmit/1357497548171907072

The way that we handle it is add retry logic to refresh the API call.

https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/argo_client.py#L87-L105

**Comment by yuzisun** [comment]
@pvaneck Any comments?

File: docs/README.md [hunk 4]
```diff
@@ -23,16 +25,35 @@ KFServing has a standardized prediction workflow across all model frameworks.
...
-# Data Plane (V2)
+### Data Plane (V2)
 The second version of the data-plane protocol addresses several issues found with the V1 data-plane protocol, including performance and generality across a large number of model frameworks and servers.
 
-## Predict
+#### Predict
 The V2 protocol proposes both HTTP/REST and GRPC APIs. See the [complete proposal](/docs/predict-api/v2) for more information.
+
+## Control Plane
+KFServing Control Plane: Responsible for reconciling the `InferenceService` and `TrainedModel` custom resources. Addtionally it creates the serverless deployment for predictor, transformer, explainer to enable
```
Comment (pvaneck) [inline docs/README.md:43]: nit: Typo s/Addtionally/Additionally/

**Review by pvaneck — COMMENTED** [review]
I tested out the 0.5-rc2 release on both OpenShift and IKS and KFServing seemed to run fine. Had some issues with multi model serving, but I think those issues were resolved in this release. So, I think this is good to go.

One question: Should the main README now (or soon) start using the v1beta1 API in the `Test KFServing Installation` section to encourage its usage? I guess one caveat is that the v1beta1 sklearn example uses the v2 predict protocol which may not be ready.

**Comment by yuzisun** [comment]
> I tested out the 0.5-rc2 release on both OpenShift and IKS and KFServing seemed to run fine. Had some issues with multi model serving, but I think those issues were resolved in this release. So, I think this is good to go.
> 
> One question: Should the main README now (or soon) start using the v1beta1 API in the `Test KFServing Installation` section to encourage its usage? I guess one caveat is that the v1beta1 sklearn example uses the v2 predict protocol which may not be ready.

right, can switch to use tensorflow example actually, will update in a separate PR then.

**Comment by yuzisun** [comment]
@animeshsingh @pvaneck @cliveseldon  0.5 is good to go, please approve

**Review by pvaneck — APPROVED** [review]
/lgtm

**Comment by animeshsingh** [comment]
Thanks @yuzisun 

/lgtm

**Pull request merged**